internal/graph/storagewrapper.go

```diff
@@ -75,17 +74,18 @@ type CachedDatastore struct {
 
 // NewCachedDatastore returns a wrapper over a datastore that caches iterators in memory.
 func NewCachedDatastore(
-	inner storage.OpenFGADatastore,
+	inner storage.RelationshipTupleReader,
 	cache storage.InMemoryCache[any],
 	maxSize int,
 	ttl time.Duration,
+	sf *singleflight.Group,
 ) *CachedDatastore {
 	return &CachedDatastore{
-		OpenFGADatastore: inner,
-		cache:            cache,
-		maxResultSize:    maxSize,
-		ttl:              ttl,
-		sf:               &singleflight.Group{},
```

Comment:
the server will hold a reference to the `sfGroup` so it can be shared across requests, but we add the ability of calling `NewCachedDatastore` per request.

Reply:
Uh? It was already shared across requests as this is not done per request, but at the boot of the service.

Reply:
Yes exactly, right now `NewCachedDatastore` is instantiated once at the server level, so the `sf` object is reused across requests, but with this PR i want all datastore wrappers to be able to be instantiated per-request, so i need to pass in the shared `sf`.